libkbfs/folder_branch_ops.go

```diff
@@ -1782,34 +1782,38 @@ func (fbo *folderBranchOps) isRevisionConflict(err error) bool {
 }
 
 func (fbo *folderBranchOps) finalizeMDWriteLocked(ctx context.Context,
-	lState *lockState, md *RootMetadata, bps *blockPutState) (err error) {
+	lState *lockState, md *RootMetadata, bps *blockPutState, excl EXCL) (err error) {
 	fbo.mdWriterLock.AssertLocked(lState)
 
 	// finally, write out the new metadata
 	mdops := fbo.config.MDOps()
 
-	doUnmergedPut, wasMasterBranch := true, fbo.isMasterBranchLocked(lState)
+	doUnmergedPut, wasMasterBranch := (excl == NoEXCL), fbo.isMasterBranchLocked(lState)
 	mergedRev := MetadataRevisionUninitialized
 
 	if fbo.isMasterBranchLocked(lState) {
 		// only do a normal Put if we're not already staged.
 		err = mdops.Put(ctx, md)
-		doUnmergedPut = fbo.isRevisionConflict(err)
-		if err != nil && !doUnmergedPut {
-			return err
-		}
-		// The first time we transition, our last known MD revision is
-		// the same (at least) as what we thought our new revision
-		// should be.  Otherwise, just leave it at uninitialized and
-		// let the resolver sort it out.
-		if doUnmergedPut {
+
+		if doUnmergedPut = fbo.isRevisionConflict(err); doUnmergedPut {
 			fbo.log.CDebugf(ctx, "Conflict: %v", err)
 			mergedRev = md.Revision
+
+			if excl == WithEXCL {
```

Comment:
Hmm.. my understanding is that, it's possible that local revision is not sync'ed with server yet, so it's possible that previous check (will be in #109) passes, but the `Put()` results in a conflict. In this case, if `O_EXCL` is set, we should not continue with unmerged Put right?

Are we gonna have any other operation that supports `O_EXCL` or is this a pretty specialized scenario? If latter, maybe we could probably make `EXCL` carry information regarding what ops call this is and only honor it if it's from `CreateFile`?